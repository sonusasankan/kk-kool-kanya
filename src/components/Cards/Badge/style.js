import styled from 'styled-components';

export const BadgeWrapper = styled.div`
    ${props => ((props.top || props.right || props.bottom || props.left) ? 'position: absolute' : '')};
    ${props => (props.top ? 'top: 1rem' : '')};
    ${props => (props.right ? 'right: 1rem' : '')};
    ${props => (props.bottom ? 'bottom: 1rem' : '')};
    ${props => (props.left ? 'left: 1rem' : '')};
    z-index: 2;
`;

export const Badge = styled.div`
    display: inline-block;
    height: 24px;
    width: 24px;  
    position: relative;
    img{
        width: 100%;
    }
    margin-left: -10px;
    &:nth-child(1){
        margin-left: 0;
    }
`;
