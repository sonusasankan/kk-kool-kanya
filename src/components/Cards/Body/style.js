import styled from 'styled-components';

const Body = styled.div`
  padding: 1.5rem;
  background-color: #fff;
  width: 100%;
  &.overlay{
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background-color: rgba(0,0,0,0);
    padding: 0;
    display: flex;
    align-items: flex-end;
    .kk-card__text-wrapper{
      padding:  16px !important;
      background-color: rgba(0,0,0,0)!important;
      color: #fff;
      p{
        margin-bottom: 0;
      }
    }
    .kk-card__title{
      margin: 0;
      font-size: 1.125rem;
    }
  }
`;

export default Body;
