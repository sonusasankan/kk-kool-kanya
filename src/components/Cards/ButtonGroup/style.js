import styled from 'styled-components';

const ButtonGroup = styled.div`
    display: flex;
    margin: 0 -16px;
    position: relative;
    bottom: -8px;
    border-radius: 0 0 4px 4px;
    overflow: hidden;
    button{
        flex-grow: 1;
        display: block;
        margin: 0;
        border-radius: 0;
        font-size: 12px;
        padding: 8px 0;
        text-align: center;
        min-height: 34px;
        ${props => (props.split ? `max-width: ${100 / props.split}%` : '')};
    }
`;

export default ButtonGroup;
