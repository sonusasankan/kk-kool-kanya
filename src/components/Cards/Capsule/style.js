import styled from 'styled-components';
import Color from '../../../themes/color';

const Capsule = styled.div`
    height: 72px;
    border-radius: 36px;
    padding: 16px;
    display: inline-flex;
    align-items: center;
    font-height: 1.125rem;
    background-color: #efeff5;
    cursor: pointer;
    .capsule{
        &__title{
            margin-left: 1rem;
        }
        &__img__wrapper{
            height: 44px;
            width: 44px;
            border-radius: 44px;
            overflow: hidden;
        }
    }
    &.selected{
        background-color: ${Color.brandColor};
        color: #FFF;
    }
`;

export default Capsule;