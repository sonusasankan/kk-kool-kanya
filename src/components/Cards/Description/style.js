import styled from 'styled-components';

import Color from '../../../themes/color';

const Description = styled.p`
    margin-top: 16px;
    margin-bottom: 0;
    line-height: 1.44;
    color: #4f565e;
    font-size: 1rem;
    .readMore{
        display: inline;
        color: ${Color.brandColor};
        font-weight: 600;
        cursor: pointer;
    }    
`;

export default Description;
