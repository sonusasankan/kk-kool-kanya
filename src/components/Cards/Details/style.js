import styled from 'styled-components';

import Color from '../../../themes/color';


export const DetailsWrapper = styled.div`
    display: flex;
    margin-bottom: 8px;
    align-items: center;
    // justify-content: ${props => (props.center? "center": "left" )}
`;

export const Details = styled.div`
    color: ${Color.fontLight};
    font-size: 0.875rem;
    text-align: ${props => (props.center? "center": "left" )}
`;

export const Icon = styled.span`
    color: ${Color.fontLight};
    margin-right: 4px;
`;
