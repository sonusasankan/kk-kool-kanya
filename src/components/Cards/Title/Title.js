import React from 'react';
import PropTypes from 'prop-types';

import Title from './style';
import getExcerpt from '../getExcerpt';

const cardTitle = ({
  excerptLength,
  children,
}) => (
  <Title className="kk-card__title">{getExcerpt(children, excerptLength)}{console.log(getExcerpt)}</Title>
);

cardTitle.propTypes = {
  excerptLength: PropTypes.number,
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.string]).isRequired
};

cardTitle.defaultProp = {
  excerptLength: 8
}


export default cardTitle; 
