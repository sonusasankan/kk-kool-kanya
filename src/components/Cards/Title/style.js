import styled from 'styled-components';

const Title = styled.h3`
  color: #292929;
  flex-grow: 1;
  font-size: 1rem;
  font-weight: 500;
  font-family: "TisaOt";
  line-height: 1.25;
  overflow: hidden;

  &.big{
    font-size: 1.5rem;
  }
`;

export default Title;
