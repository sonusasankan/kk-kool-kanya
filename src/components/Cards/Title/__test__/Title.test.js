import React from 'react';
import { shallow } from 'enzyme';

import { default as CardTitle } from '../Title';

// import Logo from '../assets/images/godrej.png';

import '../../../../assets/icons/style.css';

describe('Component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <CardTitle
        logo="https://loremflickr.com/260/220"
        rating="4.2"
      >
Godrej Consumer Products Ltd
      </CardTitle>,
    );
    expect(component).toMatchSnapshot();
  });
});
