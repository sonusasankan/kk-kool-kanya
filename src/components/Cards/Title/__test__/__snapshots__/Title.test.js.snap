// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <cardTitle
    alt=""
    icon={false}
    logo="https://loremflickr.com/260/220"
    rating="4.2"
  >
    Godrej Consumer Products Ltd
  </cardTitle>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <ForwardRef>
          <img
            alt=""
            className="w-100"
            src="https://loremflickr.com/260/220"
          />
        </ForwardRef>,
        <ForwardRef
          className="flex-grow-1 kk-card__title"
        >
          Godrej Consumer Products Ltd
        </ForwardRef>,
        <ForwardRef
          className="card__rating mx-2"
        >
          4.2
        </ForwardRef>,
      ],
      "className": "d-flex mb-3",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <img
            alt=""
            className="w-100"
            src="https://loremflickr.com/260/220"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "w-100",
            "src": "https://loremflickr.com/260/220",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  width: 24px;
  min-width: 24px;
  height: 24px;
  min-height: 24px;
  border-radius: 24px;
  margin-right: 8px;
  overflow: hidden;
  box-shadow: 0px 0px 0px 2px rgba(238,239,242,0.52);
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Godrej Consumer Products Ltd",
          "className": "flex-grow-1 kk-card__title",
        },
        "ref": null,
        "rendered": "Godrej Consumer Products Ltd",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  font-size: 1rem;
  font-weight: 500;
  margin-bottom: 0;
  font-family: \\"TisaOt\\";
  line-height: 1.25rem;
  padding-top: 4px;
",
            ],
          },
          "displayName": "styled.h3",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "h3",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "4.2",
          "className": "card__rating mx-2",
        },
        "ref": null,
        "rendered": "4.2",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  font-weight: 600;
  color: ",
              "#80c888",
              "
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <ForwardRef>
            <img
              alt=""
              className="w-100"
              src="https://loremflickr.com/260/220"
            />
          </ForwardRef>,
          <ForwardRef
            className="flex-grow-1 kk-card__title"
          >
            Godrej Consumer Products Ltd
          </ForwardRef>,
          <ForwardRef
            className="card__rating mx-2"
          >
            4.2
          </ForwardRef>,
        ],
        "className": "d-flex mb-3",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <img
              alt=""
              className="w-100"
              src="https://loremflickr.com/260/220"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "w-100",
              "src": "https://loremflickr.com/260/220",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  width: 24px;
  min-width: 24px;
  height: 24px;
  min-height: 24px;
  border-radius: 24px;
  margin-right: 8px;
  overflow: hidden;
  box-shadow: 0px 0px 0px 2px rgba(238,239,242,0.52);
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Godrej Consumer Products Ltd",
            "className": "flex-grow-1 kk-card__title",
          },
          "ref": null,
          "rendered": "Godrej Consumer Products Ltd",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  font-size: 1rem;
  font-weight: 500;
  margin-bottom: 0;
  font-family: \\"TisaOt\\";
  line-height: 1.25rem;
  padding-top: 4px;
",
              ],
            },
            "displayName": "styled.h3",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "h3",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "4.2",
            "className": "card__rating mx-2",
          },
          "ref": null,
          "rendered": "4.2",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  font-weight: 600;
  color: ",
                "#80c888",
                "
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
